projects/hslayers/src/common/layers/interpolated-source.ts

```diff
@@ -0,0 +1,200 @@
+      return;
+    }
+    const featsCached = this.featureCache.getFeatures();
+    const cacheLimit = this.options.maxFeaturesInCache;
+    if (cacheLimit < featsCached.length + features.length) {
+      const cntToRemove = featsCached.length - (cacheLimit - features.length);
+      featsCached
+        .slice(0, cntToRemove)
+        .forEach((f) => this.featureCache.removeFeature(f));
+    }
+    const countToAdd = cacheLimit ?? Number.MAX_VALUE;
+    this.featureCache.addFeatures(features.slice(0, countToAdd));
+    this.normalizeWeight(this.options.weight);
+    const src = super.getSource();
+    if (extent) {
+      src.clear();
+      const limitInExt = this.options.maxFeaturesInExtent ?? Number.MAX_VALUE;
+      src.addFeatures(
+        this.featureCache.getFeaturesInExtent(extent).slice(0, limitInExt)
+      );
+    } else {
+      src.addFeatures(features);
+    }
+  }
+
+  /**
+   * Parse features from get request
+   * @param collection - Get request response feature collection
+   * @param mapProjection - Map projection
+   */
+  parseFeatures(collection: any, mapProjection: string | Projection): void {
+    const dataProj = (collection.crs || collection.srs) ?? 'EPSG:4326';
+    collection.features = collection.features.filter(
+      (f) => !this.geoJSONFeatures.includes(f)
+    );
+    this.geoJSONFeatures = this.geoJSONFeatures.concat(collection.features);
+    collection.features = new GeoJSON().readFeatures(collection, {
+      dataProjection: dataProj,
+      featureProjection: mapProjection,
+    });
+    collection.features = collection.features.filter((f) => {
+      const value = f.get(this.options.weight);
+      if (value && !isNaN(parseInt(value))) {
+        return f;
+      }
+    });
+    return collection.features;
+  }
+
+  /**
+   * Create url for get request including current map extent
+   * @param url - external source URL
+   * @param extent - Current map extent
+   */
+  createIDWSourceUrl(url: string, extent: number[]): string {
+    if (!url) {
+      return;
+    } else if (extent) {
+      const extentObj = [
+        {ref: 'minx', value: extent[0].toFixed(1)},
+        {ref: 'miny', value: extent[1].toFixed(1)},
+        {ref: 'maxx', value: extent[2].toFixed(1)},
+        {ref: 'maxy', value: extent[3].toFixed(1)},
+      ];
+      const matches = url.match(/{.+?}/g);
+      if (matches?.length > 0 && matches?.length <= 4) {
+        for (const m of matches) {
+          const ix = matches.indexOf(m);
+          const key = m.replace(/[{}]/g, '').toLowerCase();
+          const coord = extentObj.find((e) => e.ref === key) ?? extentObj[ix];
+          url = url.replace(m, coord.value);
+        }
+      }
+    }
+    return url;
+  }
+  /**
+   * Normalize weight values to be between 0 and 100
+   * @param features - OL feature array
+   * @param weight - Weight property name
+   */
+  normalizeWeight(weight: string): void {
+    //https://www.statology.org/normalize-data-between-0-and-100/
```

Comment:
why not moving this line into the tsdoc comment above?

Reply:
No problem! :)